File: core 2/src/main/java/com/dev/core/service/TeamService.java
package com.dev.core.service;

import com.dev.core.model.TeamDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface TeamService {

    TeamDTO createTeam(TeamDTO dto);

    TeamDTO updateTeam(Long id, TeamDTO dto);

    void deleteTeam(Long id);

    TeamDTO getTeamById(Long id);

    List<TeamDTO> getAllTeams(Long organizationId);

    Page<TeamDTO> searchTeams(Long organizationId, String keyword, Pageable pageable);

    // 🔹 Manage Members
    void addMember(Long teamId, Long employeeId, boolean isLead, boolean isManager);

    void removeMember(Long teamId, Long employeeId);

    void setTeamLead(Long teamId, Long employeeId);

    void changeManager(Long teamId, Long managerId);

    // 🔹 Analytics
    long getTeamSize(Long teamId);

    List<TeamDTO> getTeamsByDepartment(Long organizationId, Long departmentId);
}

File: core-platform/apps/core-webapp/src/services/team.service.ts
import axiosInstance from "../axiosInstance";
import type {
  TeamDTO,
  TeamPageResponse,
  CreateTeamRequest,
  UpdateTeamRequest,
} from "../types/team.types";

const TEAM_API_BASE = "/teams";

export const teamService = {
  // Create a new team
  createTeam: async (data: CreateTeamRequest): Promise<TeamDTO> => {
    const response = await axiosInstance.post(TEAM_API_BASE, data);
    return response.data.data;
  },

  // Update an existing team
  updateTeam: async (id: number, data: UpdateTeamRequest): Promise<TeamDTO> => {
    const response = await axiosInstance.put(`${TEAM_API_BASE}/${id}`, data);
    return response.data.data;
  },

  // Delete a team
  deleteTeam: async (id: number): Promise<void> => {
    await axiosInstance.delete(`${TEAM_API_BASE}/${id}`);
  },

  // Get team by ID
  getTeamById: async (id: number): Promise<TeamDTO> => {
    const response = await axiosInstance.get(`${TEAM_API_BASE}/${id}`);
    return response.data.data;
  },

  // Get all teams for an organization
  getAllTeams: async (organizationId: number): Promise<TeamDTO[]> => {
    const response = await axiosInstance.get(TEAM_API_BASE, {
      params: { organizationId },
    });
    return response.data;
  },

  // Search teams with pagination
  searchTeams: async (
    organizationId: number,
    keyword?: string,
    page: number = 0,
    size: number = 20,
  ): Promise<TeamPageResponse> => {
    const response = await axiosInstance.get(`${TEAM_API_BASE}/search`, {
      params: {
        organizationId,
        keyword,
        page,
        size,
        sort: "name,asc",
      },
    });
    return response.data.data;
  },

  // Add a member to a team
  addMember: async (
    teamId: number,
    employeeId: number,
    isLead: boolean = false,
    isManager: boolean = false,
  ): Promise<void> => {
    await axiosInstance.post(`${TEAM_API_BASE}/${teamId}/members`, null, {
      params: { employeeId, isLead, isManager },
    });
  },

  // Remove a member from a team
  removeMember: async (teamId: number, employeeId: number): Promise<void> => {
    await axiosInstance.delete(
      `${TEAM_API_BASE}/${teamId}/members/${employeeId}`,
    );
  },

  // Set team lead
  setTeamLead: async (teamId: number, employeeId: number): Promise<void> => {
    await axiosInstance.put(`${TEAM_API_BASE}/${teamId}/lead/${employeeId}`);
  },

  // Change team manager
  changeManager: async (teamId: number, managerId: number): Promise<void> => {
    await axiosInstance.put(`${TEAM_API_BASE}/${teamId}/manager/${managerId}`);
  },
};

export default teamService;
